seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpPostCommandProcessor.java

```diff
@@ -96,14 +98,20 @@ private void handleSubmitJob(HttpPostCommand httpPostCommand, String uri)
         Config config = RestUtil.buildConfig(requestHandle(httpPostCommand));
         JobConfig jobConfig = new JobConfig();
         jobConfig.setName(requestParams.get(RestConstant.JOB_NAME));
+        if (Common.getDeployMode() == null) {
+            Common.setDeployMode(DeployMode.CLIENT);
+        }
+        boolean startWithSavePoint =
+                Boolean.parseBoolean(requestParams.get(RestConstant.IS_START_WITH_SAVE_POINT));
         RestJobExecutionEnvironment restJobExecutionEnvironment =
                 new RestJobExecutionEnvironment(
                         jobConfig,
                         config,
                         textCommandService.getNode(),
-                        Boolean.parseBoolean(
-                                requestParams.get(RestConstant.IS_START_WITH_SAVE_POINT)),
-                        Long.parseLong(requestParams.get(RestConstant.JOB_ID)));
+                        startWithSavePoint,
+                        startWithSavePoint
+                                ? Long.parseLong(requestParams.get(RestConstant.JOB_ID))
+                                : null);
```

Comment:
Can we report error more clear if we start with savepoint but without put jobId in params. Seem like it will report NPE at now. I think the error message should more clear.

Reply:
added parameter check